### ISSUE TITLE
BLE does not reconnect after disconnect version 1.05 and up

### ISSUE THREAD

**Issue by Zparq** [espressif/arduino-esp32#6016]
Using example code provided for BLE Server, the code compiles and runs. However, with version 1.05 and newer, it cannot reconnect BLE after a disconnect event initiated from an Android device. For example if I pair the ESP32 with the phone and connect, everything works fine. I then disable the bluetooth on the android for 15 seconds and then reenable it. After this it is impossible to reconnect to the ESP32. However it is still working with version 1.04.

**Comment by hyperlite2** [comment]
I can confirm that this is the case. 1.0.4 works well to reconnect BT after disconnect, but all future versions seem to break the ability to reconnect.

**Comment by Dork57** [comment]
I can confirm, too. No advertising after disconnect.

**Comment by me-no-dev** [comment]
have you tried restarting the advertisement after disconnect?

**Comment by Dork57** [comment]
> have you tried restarting the advertisement after disconnect?

No, but you're right. Thanks. I just found the solution also in the example code of the BLE Lib.
`pServer->startAdvertising(); // restart advertising after disconnecting`

**Comment by VojtechBartoska** [comment]
Hello, can you please retest this on v2.0.3-rc1?

**Comment by RodrigoALippi** [comment]
Hi!!
 By Espressif System 2.0.3-RC1...

class MyServerCallbacks: public BLEServerCallbacks {
    void onConnect(BLEServer* pServer) {
      deviceConnected = true;
    };
 
    void onDisconnect(BLEServer* pServer) {
      deviceConnected = false;
      pServer->getAdvertising()->start();
    }
};

Enjoy!

**Comment by VojtechBartoska** [comment]
We will take a look on this, adding to our Roadmap.

**Comment by gentigace** [comment]
@RodrigoALippi many thanks man this solve also my issue.
